tokens/delivery/http/tokens_delivery.go

```diff
@@ -123,11 +123,16 @@ func (a *TokensHandler) GetMetadata(c echo.Context) (err error) {
 }
 
 // @Summary Get prices
-// @Description Given a list of base denominations, returns the spot price with a system-configured quote denomination.
+// @Description Given a list of base denominations, this endpoint returns the spot price with a system-configured quote denomination.
+// If the pricing source is set to "chain" (0), it will first check the **chain** pricing cache for the price quote. If it exists, it will return it. Otherwise, it will compute the pricing on-demand if the quote is non-usdc.
+// If the pricing source is set to "coingecko" (1), it will look for the price quote in the **coingecko** pricing cache. If it exists, it will return it. Otherwise, it will fetch the price from the Coingecko API endpoint and store it in the cache with an expiration time specified in the config.json file.
+// If the token price is not available from the chain pricing source for any reason, it will fallback to the Coingecko pricing source if the quote denomination (human or chain) is usdc.
+// See also: https://github.com/osmosis-labs/sqs/blob/de34d172f95b221217967799f233c52181cfa07e/README.md#pricing
```

Comment:
Nice! Could you please also extend the CoinGecko section documentation here:
https://github.com/osmosis-labs/sqs/blob/v24.x/README.md#coingecko

In a subsequent effort, I propose we move some of this detail under `docs/architecture/pricing.md` to avoid having the README grow out of control.

Instead, we can link specific documents(such as `pricing.md`) from the top level README

Reply:
@p0mvn , can you share me the link to pricing.md ? I don't seem to find it in https://github.com/osmosis-labs/docs/tree/main Thanks!